language-plutus-core/src/Language/PlutusCore/Size.hs

```diff
@@ -0,0 +1,36 @@
+module Language.PlutusCore.Size ( termSize
+                                ) where
+
+import           Data.Functor.Foldable
+import           Language.PlutusCore.Type
+
+kindSize :: Kind a -> Int
+kindSize = cata a where
+    a TypeF{}             = 1
+    a (KindArrowF _ k k') = 1 + k + k'
+    a SizeF{}             = 1
+
+typeSize :: Type tyname a -> Int
+typeSize = cata a where
+    a TyVarF{}             = 1
+    a (TyFunF _ ty ty')    = 1 + ty + ty'
+    a (TyIFixF _ ty ty')   = 1 + ty + ty'
+    a (TyForallF _ _ k ty) = 1 + kindSize k + ty
+    a TyBuiltinF{}         = 1
+    a TyIntF{}             = 1
+    a (TyLamF _ _ k ty)    = 1 + kindSize k + ty
+    a (TyAppF _ ty ty')    = 1 + ty + ty'
+
+-- | This exists so that we can measure term size
```

Comment:
Well, yes. Perhaps a more useful doc would say what it actually does? In this case something like "counts the number of AST nodes in a term"?